src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCISharedBuildJobQueueService.java

```diff
@@ -104,35 +111,90 @@ public List<LocalCIBuildJobQueueItem> getProcessingJobsForCourse(long courseId)
         return processingJobs.values().stream().filter(job -> job.getCourseId() == courseId).toList();
     }
 
-    /**
-     * Get first build job item from the queue. If it exists, process build job and after completion,
-     * try to process next item.
-     */
-    private void processBuild() {
+    private ProgrammingExerciseParticipation retrieveParticipationWithRetry(Long participationId) {
+        int maxRetries = 5;
+        int retries = 0;
+        while (retries < maxRetries) {
+            try {
+                return (ProgrammingExerciseParticipation) participationRepository.findByIdElseThrow(participationId);
+            }
+            catch (Exception e) {
+                log.debug("Error while retrieving participation with id " + participationId + " from database: " + e.getMessage());
+                log.info("Retrying to retrieve participation with id " + participationId + " from database");
+                retries++;
+                try {
+                    Thread.sleep(1000);
+                }
+                catch (InterruptedException e1) {
+                    log.error("Error while waiting for participation with id " + participationId + " to be persisted in database: " + e1.getMessage());
```

Comment:
```suggestion
                    log.error("Error while waiting for participation with id {} to be persisted in database.", participationId, e1);
```
Warn and error logs can take the exception directly to handle the error message logging automatically.